src/clib/pio_msg.c

```diff
@@ -1638,13 +1796,19 @@ int delete_att_handler(iosystem_desc_t *ios)
     if (!(name = malloc((namelen + 1) * sizeof(char))))
         return pio_err(ios, NULL, PIO_ENOMEM, __FILE__, __LINE__);
     if ((mpierr = MPI_Bcast(name, namelen + 1, MPI_CHAR, 0, ios->intercomm)))
+    {
+        free(name);
         return check_mpi2(ios, NULL, mpierr, __FILE__, __LINE__);
+    }
     LOG((2, "delete_att_handler namelen = %d name = %s ncid = %d varid = %d ",
          namelen, name, ncid, varid));
 
     /* Call the create file function. */
     if ((ret = PIOc_del_att(ncid, varid, name)))
+    {
+        free(name);
         return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+    }
```

Comment:
I suspect there is a better way to handle this issue than just adding free(name) everywhere. How about a sole exit point for errors where it is freed? Or how about we use a declared instead of mallocd name value? If declared at size PIO_MAX_NAME + 1 it will always be long enough and does not have to be freed everywhere.